projects/plugins/jetpack/modules/contact-form/admin.php

```diff
@@ -1128,3 +1113,263 @@ function grunion_feedback_admin_notice() {
 	}
 }
 add_action( 'admin_notices', 'grunion_feedback_admin_notice' );
+
+/**
+ * Class Grunion_Admin
+ *
+ * Stop the global scope madness, use static methods on this class to provide for wp-admin responses area.
+ * Might turn into singleton in the future.
+ */
+class Grunion_Admin {
+	/**
+	 * Define nonce field name
+	 *
+	 * @var string The nonce field name.
+	 */
+	private $export_nonce_field = 'feedback_export_nonce';
+
+	/**
+	 * Instantiates this singleton class
+	 *
+	 * @return Grunion_Admin The Grunion Admin class instance.
+	 */
+	public static function init() {
+		static $instance = false;
+
+		if ( ! $instance ) {
+			$instance = new Grunion_Admin();
+		}
+
+		return $instance;
+	}
+
+	/**
+	 * Grunion_Admin constructor
+	 */
+	public function __construct() {
+		if ( defined( 'JETPACK_BETA_BLOCKS' ) && JETPACK_BETA_BLOCKS ) {
+			add_action( 'admin_enqueue_scripts', array( $this, 'admin_enqueue_scripts' ) );
+			add_action( 'admin_footer-edit.php', array( $this, 'print_export_modal' ) );
+			add_action( 'wp_ajax_grunion_export_to_gdrive', array( $this, 'export_to_gdrive' ) );
+		}
+	}
+
+	/**
+	 * Hook handler for admin_enqueue_scripts hook
+	 */
+	public function admin_enqueue_scripts() {
+		$current_screen = get_current_screen();
+		if ( ! in_array( $current_screen->id, array( 'edit-feedback', 'feedback_page_feedback-export' ), true ) ) {
+			return;
+		}
+		add_thickbox();
+	}
+
+	/**
+	 * Prints the modal markup with export buttons/content.
+	 */
+	public function print_export_modal() {
+		if ( ! current_user_can( 'export' ) ) {
+			return;
+		}
+
+		// if there aren't any feedbacks, bail out
+		if ( ! (int) wp_count_posts( 'feedback' )->publish ) {
+			return;
+		}
+
+		$current_screen = get_current_screen();
+		if ( ! in_array( $current_screen->id, array( 'edit-feedback', 'feedback_page_feedback-export' ), true ) ) {
+			return;
+		}
+		?>
+		<div id="feedback-export-modal" style="display: none;">
+			<div class="feedback-export-modal__wrapper">
+				<div class="feedback-export-modal__header">
+					<h1 class="feedback-export-modal__header-title">Export your Form responses</h1>
+					<p class="feedback-export-modal__header-subtitle">Choose your favorite file format or export destination:</p>
+				</div>
+				<div class="feedback-export-modal__content">
+					<?php $this->get_csv_export_section(); ?>
+					<?php $this->get_gdrive_export_section(); ?>
```

Comment:
Should we avoid displaying the section if the user is not connected to WordPress.com?

Reply:
I rely on the method to return nothing on those cases, should I bring it up into the modal output code?

```
public function get_gdrive_export_section() {
	$user_connected = ( defined( 'IS_WPCOM' ) && IS_WPCOM ) || ( new Connection_Manager( 'jetpack' ) )->is_user_connected( get_current_user_id() );
	if ( ! $user_connected ) {
		return;
	}
	...
}
```